app/controllers/repositories_controller.rb

```diff
@@ -182,4 +225,24 @@ def create_equipments
         Equipment.create(name: e, repository_id: @repository.id)
       end if params['equipments'].present?
     end
+
+    def update_share_type
```

Comment:
Do you need this method? I thought that ` @repository.update(repository_params)` would do this already?